### PR TITLE
Fix max length text fields overflowing

### PR DESCRIPTION
**Pull request by joonatank**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- fix: (admin) reservation list first load not showing loader.
- fix: (admin) reservation info text boxes overflowing.
- fix: (ui): email input in reservation form causing a backend error.

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- Follow up for https://github.com/City-of-Helsinki/tilavarauspalvelu-ui/pull/1055
- Follow up for https://github.com/City-of-Helsinki/tilavarauspalvelu-ui/pull/1044

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- [TILA-3209](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3209)

[TILA-3209]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3209?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/common/src/reservation-form/ReservationFormField.tsx [hunk 1]
```diff
@@ -445,7 +445,8 @@ const ReservationFormField = ({
           defaultValue={defaultValue ? String(defaultValue) : undefined}
           invalid={!!error}
           required={required}
-          maxLength={MAX_TEXT_LENGTH}
+          // email field is special and has one less character than the rest
+          maxLength={MAX_TEXT_LENGTH - (isEmailField ? 1 : 0)}
```
Comment (ranta) [inline packages/common/src/reservation-form/ReservationFormField.tsx:449]: Why should email fields have one less character? 🤔

File: apps/admin-ui/src/component/reservations/ReservationsDataLoader.tsx [hunk 2]
```diff
@@ -129,7 +129,7 @@ const ReservationsDataLoader = ({
     sort
   );
 
-  if (loading && !data) {
+  if (loading && data.length === 0) {
```
Comment (ranta) [inline apps/admin-ui/src/component/reservations/ReservationsDataLoader.tsx:132]: is `data` ever `undefined` or `null`?
(Do we need to check for both here: `(!data || data.length === 0)` or `data?.length === 0`?)

**Review by ranta — APPROVED** [review]

File: apps/admin-ui/src/component/reservations/ReservationsDataLoader.tsx (hunk 2 above)
Comment (joonatank) [reply apps/admin-ui/src/component/reservations/ReservationsDataLoader.tsx:132]: It's never null (goes through a filter function that returns [] for null). 
Typescript would raise an error for `data.length === 0` and also `data?.length === 0`

File: packages/common/src/reservation-form/ReservationFormField.tsx (hunk 1 above)
Comment (joonatank) [reply packages/common/src/reservation-form/ReservationFormField.tsx:449]: Good question, something weird in the backend constraints. 255 for all other fields, 254 for email 🙃  
The GQL field error specifies the max length.

**Pull request merged**